packages/labs/analyzer/src/lib/lit-element/lit-element.ts

```diff
@@ -31,26 +32,40 @@ export const getLitElementDeclaration = (
     node,
     reactiveProperties: getProperties(node, analyzer),
     events: getEvents(node, analyzer),
+    getHeritage: () => getHeritage(node, analyzer),
   });
 };
 
 /**
  * Returns true if this type represents the actual LitElement class.
  */
-const _isLitElementClassDeclaration = (t: ts.BaseType) => {
+const _isLitElementClassDeclaration = (
+  t: ts.BaseType,
+  analyzer: AnalyzerInterface
+) => {
   // TODO: should we memoize this for performance?
   const declarations = t.getSymbol()?.getDeclarations();
   if (declarations?.length !== 1) {
     return false;
   }
   const node = declarations[0];
+  return _isLitElement(node) || isLitElementSubclass(node, analyzer);
+};
+
+/**
+ * Returns true if the given declaration is THE LitElement declaration.
+ */
+const _isLitElement = (node: ts.Declaration) => {
   return (
     _isLitElementModule(node.getSourceFile()) &&
     ts.isClassDeclaration(node) &&
     node.name?.text === 'LitElement'
   );
 };
 
+/**
+ * Returns true if the given source file is THE lit-element source file.
```

Comment:
Seems potentially brittle since it depends on building from canonical source.

Reply:
Yeah. I added a TODO.

If someone is depending on a non-canonical Lit (bundled or mangled in some unknown way), we could only support that by trying to shape-check the class, which is also going to be brittle. I don't think we need to go there anytime soon.